modules/docs/arrow-docs/docs/docs/arrow/data/mapk/README.md

```diff
@@ -11,13 +11,82 @@ redirect_from:
+```
+
+Sometimes you need to "map" one Map to another. You are able to do so with `map2(..)` function:
+```kotlin:ank
+    val firstBag = mapOf("eggs" to 5, "milk" to 1).k()
+    val secondBag = mapOf("eggs" to 6, "cheese" to 1).k()
+
+    val eggsBag = firstBag.map2(secondBag) { firstBagMatch, secondBagMatch ->     // If there are matching keys
+        firstBagMatch + secondBagMatch     // you can modify the value of output MapK
+    }
+    // eggsBag = MapK(map={eggs=11})
 ```
+`map2Eval` does pretty much the same as `map2`, but result `KMap` will be wrapped in [`Eval`](https://arrow-kt.io/docs/arrow/core/eval/#eval) type.
+
+`ap` function is used when you want to apply map of transformations from `Map<K, (A)-> B>` to `Map<K,A>`, for example:
+```kotlin:ank
+    val map1: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
+    
+    val f1 = { i: Int -> "f1 to \"$i\"" }
+    val f2 = { i: Int -> "f2 to \"$i\"" }
+    val map2: MapK<String, (Int) -> String> = mapOf("one" to f1, "two" to f2).k()
+
+    val apResult = map1.ap(map2) // MapK(map={one=f1 to "1", two=f2 to "2"})
+``` 
+`ap2` acts like `map2` to `map`
+
+In most cases you would like to use `flatMap` function which flattens source map, accepts `(A) -> MapK<K,B>` functor and returns `MapK<K,Z>`
+```kotlin:ank
+    val map1: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
+    val map2: MapK<String, String> = mapOf("one" to "ONE", "three" to "three").k()
+
+    val flattened = map1.flatMap { map2 } // MapK(map={one=ONE})
+```
+
+`foldLeft` and `foldRight` are used for element aggregation:
+```kotlin:ank
+    val map1: MapK<String, Int> = mapOf("one" to 1, "two" to 2).k()
+
+    val foldLeft: String = map1.foldLeft("one") { entry, tuple -> entry + tuple} // "one12"
+    val foldRight: Eval<String> =                                                // foldRight.value() = "1 2 one", since it is Eval
+        map1.foldRight(Eval.just("one")) { entry, eval ->
+            Eval.just("$entry ${eval.value()}")
+        }
+```
+
+You can also traverse `MapK` data structure performing an action on each element.
+```kotlin:ank
+import arrow.core.None
+import arrow.core.Option
+import arrow.core.Some
+import arrow.core.extensions.option.applicative.applicative
+import arrow.core.fix
+import arrow.core.k
+import arrow.core.some
 
-TODO. Meanwhile you can find a short description in the [intro to datatypes]({{ '/docs/datatypes/intro/' | relative_url }}).
+    val optionMap = mapOf(1.some() to "one", 2.some() to "two", None to "none").k()
+            .traverse(Option.applicative()) { value ->
+                when (value) {
+                    "one", "two", "none" -> Some(value)
+                    else -> None
+                }
+            }.fix()
+        println(optionMap) // Some(MapK(map={Some(1)=one, Some(2)=two, None=none}))
+``` 
+TODO: add link to `Traverse` docs when it's ready https://github.com/arrow-kt/arrow/pull/1534
```

Comment:
Remaining `TODO`?

Reply:
Yes, because @i-walker is working on it.

Reply:
@rcd27 
- please check your formating in all snippets- in Arrow ident's are `2` everywhere
- playground snippets only run what's defined in `main()`